src/prevector.h

```diff
@@ -360,6 +360,7 @@ class prevector
         }
     }
 
+    prevector(prevector<N, T, Size, Diff> &&other) : _size(0) { swap(other); }
     prevector &operator=(const prevector<N, T, Size, Diff> &other)
```

Comment:
what do these prevector changes have to do with random numbers or did they just slip in during cherry-picking?

Reply:
This come from bitcoin/bitcoin/pull/9349 changes `src/test/prevectors_test.cpp`, some file has been modified by the proper random cherry pick I made, so I decided to port also this PR. 

In hindsight I could have left it out cause it is not related to the focus of this PR.